demonstrations/tutorial_unitary_synthesis_kak.py

```diff
@@ -0,0 +1,512 @@
+with the setting ``separate=True``, then computes the blocks for the block-diagonal matrices as well
+as the coefficients, or rotation angles, for :math:`A\in\exp(\mathfrak{a}_{\text{AIII}})`.
+
+[CODE]: Could add an implementation, drawing from QSD implementation in PL and/or KAK tools repo.
+
+If a different Cartan subalgebra is chosen, a suitable rotation needs to be applied. This is
+usually a matter of right(left)-multiplying :math:`K_1` (:math:`K_2`) by the rotation and
+re-interpreting the obtained angles to parametrize the new Cartan subgroup element already.
+
+[EXAMPLE?]
+
+Type-A decomposition
+--------------------
+
+A type-A Cartan decomposition is defined for the "doubled" unitary group 
+:math:`\mathcal{G}=U(d)^{\times 2}` and its subgroup :math:`\mathcal{K}=U(d)`.
+As you may anticipate, we still only care about qubit space dimensions, :math:`d=2^n`, for some 
+qubit count :math:`n`.
+Generically, we may use the representation of :math:`\mathcal{G}` as block-diagonal matrices
+from above, and we will use the representation
+
+.. math::
+
+    \mathcal{K}=U(2^n)
+    \cong\left\{\begin{pmatrix} A & 0 \\ 0 & A \end{pmatrix}\bigg| A\in U(2^n)\right\}
+    =\left\{\mathbb{I}_2 \otimes A | A\in U(2^n)\right\}
+
+for the subgroup.
+Note that in a quantum circuit, such an element :math:`\mathbb{I}_2\otimes A\in\mathcal{K}` 
+simply is the gate :math:`A` acting on all but the first qubit.
+
+As :math:`\mathcal{K}` is generated by matrices that do the same in the two diagonal blocks,
+the complementary horizontal space accordingly consists of elements that do the opposite
+in the two blocks,
+
+.. math:: 
+
+    \mathfrak{p} = \mathfrak{k}^\perp
+    =\left\{\begin{pmatrix} C & 0 \\ 0 & -C \end{pmatrix}\bigg| C \in \mathfrak{u}(2^n)\right\}.
+
+A valid Cartan subalgebra is then any Abelian subalgebra of :math:`\mathfrak{p}` with dimension
+:math:`2^n`.
+A particularly simple choice that will be important for us is
+
+.. math::
+
+    \mathfrak{a}_{\text{A}}
+    =\left\{\begin{pmatrix} d & 0 \\ 0 & -d \end{pmatrix}\bigg| d \in \mathfrak{u}_{\text{diag}}(2^n)\right\},
+
+which is exactly of the form of :math:`\mathfrak{a}_{\text{AIII}}` except for a basis
+change from Pauli :math:`Y` to Pauli :math:`Z`.
+We thus may draw the complete type-A decomposition using :math:`\mathfrak{a}_{\text{A}}` as
+
+.. figure:: ../_static/demonstration_assets/unitary_synthesis_kak/A_generic_complete.png
+   :align: center
+   :width: 70%
+   :alt: Generic type-A KAK decomposition of the doubled unitary group on n-1 qubits.
+
+Implementation: Demultiplexing
+++++++++++++++++++++++++++++++
+
+As for the type-AIII KAK decomposition above, we will need a numerical method that computes the
+matrices :math:`K_1, K_2\in\mathbb{I}\otimes U(2^{n-1})` and 
+:math:`A\in\exp(\mathfrak{a}_{\text{A}})` for the type-A decomposition. Luckily, this can be done
+with a standard eigenvalue decomposition and some additional matrix manipulations [#Shende_QSD]_.
```

Comment:
would find it helpful to recall the mathematical problem we are trying to solve explicitly

```suggestion
As for the type-AIII KAK decomposition above, we will need a numerical method that computes the
matrices :math:`K_1, K_2\in\mathbb{I}\otimes U(2^{n-1})` and 
:math:`A\in\exp(\mathfrak{a}_{\text{A}})` for the type-A decomposition such that 

..math:: U \oplus V = (\mathbb{1} \otimes U_1) A (\mathbb{1} \otimes U_2).

Luckily, this can be done
with a standard eigenvalue decomposition and some additional matrix manipulations [#Shende_QSD]_.
```

this is subjective but I always find it way more easy to follow when I _see_ the equation and not just its pieces flying about

Reply:
✅